### ISSUE TITLE
Having a dict with the key "self" does not work

### ISSUE THREAD

**Issue by estyrke** [syrusakbary/snapshottest#147]
If a snapshot contains a (possibly nested) dict with a key named "self", snapshottest will crash in the SortedDict constructor with the error `TypeError: __init__() got multiple values for argument 'self'`. There doesn't seem to be an obvious solution, because whatever name is chosen for the `self` parameter in `SortedDict.__init__`, it risks colliding with the incoming dict keys. Calling it something more contrived, like `__SortedDict__self__` might mitigate the problem.

Minimal pytest example:
```python
def test_self(snapshot):
    snapshot.assert_match({"self": 1})
```

**Comment by estyrke** [comment]
Apparently, the problematic code does not exist in 0.6.0 anymore, so maybe the issue is solved already. It doesn't seem to be available on pypi yet though.

**Comment by paulmelnikow** [comment]
The latest release is 1.0.0a0 which is on PyPI. Want to give that a try?

<img width="1218" alt="Screen Shot 2020-10-24 at 9 26 09 AM" src="https://user-images.githubusercontent.com/1487036/97082906-f73dac00-15da-11eb-96e0-79f4cabce46a.png">

**Comment by estyrke** [comment]
Not sure what I did wrong last time, but indeed, 0.6.0 has fixed this issue, and 1.0.0a0 works as well.
